lib/puppet/face/facts.rb

```diff
@@ -128,18 +128,32 @@
       summary _("Show legacy facts when querying all facts.")
     end
 
+    option("--value-only") do
+      summary _("Show only the value when the action is called with a single query")
+    end
+
     when_invoked do |*args|
       options = args.pop
 
       Puppet.settings.preferred_run_mode = :agent
       Puppet::Node::Facts.indirection.terminus_class = :facter
 
+      if options[:value_only] && !args.count.eql?(1)
+        options[:value_only] = nil
+        Puppet.warning("Incorrect use of --value-only argument, the argument must be used when querying for a single fact!")
```

Comment:
The language of this warning implies that one must always use `--value-only` when asking for a single fact, which isn't correct. I can query for a single fact and _not_ use the --value-only option.

This should probably say something more like:
> Incorrect use of the --value-only argument; it can only be used when querying for a single fact

can vs. must

Reply:
Thank you for noticing this. I've changed the warning message.